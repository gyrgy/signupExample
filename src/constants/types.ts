// types
export type InputFieldType = 'text' | 'number' | 'password';

// data types
export interface UserFormDataType {
  name: string;
  role: string;
  email: string;
  password: string;
}

export interface PrivacyFormDataType {
  trayProductUpdates: boolean;
  otherProductsUpdates: boolean;
}

// store types
export interface UserFormStateType extends UserFormDataType {
  submitted: boolean;
}

export interface PrivacyFormStateType extends PrivacyFormDataType {
  submitted: boolean;
}

// root store type
export interface RootStoreType {
  userForm: UserFormStateType;
  privacyForm: PrivacyFormStateType;
}
