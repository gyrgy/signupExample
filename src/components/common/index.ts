export { default as ButtonPrimary } from 'src/components/common/buttonPrimary';
export { default as Checkbox } from 'src/components/common/checkbox';
export { default as Content } from 'src/components/common/content';
export { default as Fade } from 'src/components/common/fade';
export { default as Form } from 'src/components/common/form';
export { default as InputField } from 'src/components/common/inputField';
export { default as TabPanel } from 'src/components/common/tabPanel';