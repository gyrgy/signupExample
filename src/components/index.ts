export { default as SignUpDone } from 'src/components/signUpDone';
export { default as SignUpPrivacyForm } from 'src/components/signUpPrivacyForm';
export { default as SignUpTabMenu } from 'src/components/signUpTabMenu';
export { default as SignUpUserForm } from 'src/components/signUpUserForm';
