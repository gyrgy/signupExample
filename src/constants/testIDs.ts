export const primaryButtonTestID = 'components/constants/buttonPrimary';
export const checkboxTestID = 'components/constants/checkbox';
export const contentTestID = 'components/constants/content';
export const fadeTestID = 'components/constants/fade';
export const formTestID = 'components/constants/form';
export const inputFieldTestID = 'components/constants/form';
export const tabPanelTestID = 'components/constants/tabPanel';
export const privacyFormTestID = 'components/signUpPrivacyForm';
export const privacyFormTrayCheckboxTestID =
  'components/signUpPrivacyForm/trayCheckbox';
export const privacyFormOtherCheckboxTestID =
  'components/signUpPrivacyForm/otherCheckbox';
export const tabMenuTestID = 'components/signUpTabMenu';
export const tabMenuUserTabTestID = 'components/signUpTabMenu/userTab';
export const tabMenuPrivacyTabTestID = 'components/signUpTabMenu/privacyTab';
export const tabMenuDoneTabTestID = 'components/signUpTabMenu/doneTab';
export const userFormTestID = 'components/signUpUserForm';
export const userFormNameFieldTestID = 'components/signUpUserForm/nameField';
export const userFormRoleFieldTestID = 'components/signUpUserForm/roleField';
export const userFormEmailFieldTestID = 'components/signUpUserForm/emailField';
export const userFormPasswordFieldTestID =
  'components/signUpUserForm/passwordField';
export const signUpDoneTestID = 'components/signUpDone';
