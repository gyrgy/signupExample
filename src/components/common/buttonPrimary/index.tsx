import React from 'react';
import { Button } from '@material-ui/core';
import { primaryButtonTestID } from 'src/constants/testIDs';

interface PrimaryButtonProps {
  label: string;
  onClick: () => void;
  testID?: string;
}

const ButtonPrimary = (props: PrimaryButtonProps) => {
  const { label, onClick, testID } = props;

  return (
    <Button variant="contained" color="primary" onClick={onClick} data-testid={testID || primaryButtonTestID}>
      {label}
    </Button>
  );
};

export default ButtonPrimary;
