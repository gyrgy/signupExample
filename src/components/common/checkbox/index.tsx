import React, { ChangeEvent } from 'react';
import { FormControlLabel, Checkbox } from '@material-ui/core';
import { checkboxTestID } from 'src/constants/testIDs';

interface InputCheckboxProps {
  name: string;
  label: string;
  checked: boolean;
  onChange: (event: ChangeEvent<HTMLInputElement>) => void;
  testID?: string
}


const InputCheckbox = (props: InputCheckboxProps) => {

  const { name, label, checked, onChange, testID } = props;

  return (
    <FormControlLabel
      control={<Checkbox checked={checked} onChange={onChange} name={name} color="primary" data-testid={testID || checkboxTestID} />}
      label={label}
    />
  );
};

export default InputCheckbox;
