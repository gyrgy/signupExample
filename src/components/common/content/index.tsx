import React, { ReactNode } from 'react';
import { makeStyles, Theme } from '@material-ui/core/styles';
import { contentTestID } from 'src/constants/testIDs';

const useStyles = makeStyles((theme: Theme) => ({
  root: {
    backgroundColor: theme.palette.background.paper,
  },
}));

interface ContentProps {
  children: ReactNode;
  testID?: string;
}

const Content = (props: ContentProps) => {
  const { root } = useStyles();
  const { children, testID } = props;

  return (
    <div className={root} data-testid={testID || contentTestID}>
      {children}
    </div>
  );
};

export default Content;