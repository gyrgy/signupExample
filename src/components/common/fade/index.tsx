import React, { ReactNode } from 'react';
import { Fade } from '@material-ui/core';
import { fadeTestID } from 'src/constants/testIDs';

interface FadeEffectProps {
  fadeIn: boolean;
  timeout?: number;
  children: ReactNode;
  testID?: string;
}

const FadeEffect = (props: FadeEffectProps) => {
  const { fadeIn, timeout, children, testID } = props;

  return (
    <Fade
      in={fadeIn}
      timeout={timeout || 300}
      data-testid={testID || fadeTestID}
    >
      <div>
        {children}
      </div>
    </Fade>
  );
};

export default FadeEffect;