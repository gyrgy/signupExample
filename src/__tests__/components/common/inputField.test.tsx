import React from 'react';
import { render, RenderResult, fireEvent } from '@testing-library/react';
import { InputField } from 'src/components/common';
import { inputFieldTestID } from 'src/constants/testIDs';
import { InputFieldType } from 'src/constants/types';

describe('test input field component', () =>{
  let type: InputFieldType ='text';
  const label = 'label';
  const name = 'name';
  const spyOnChangeHandler = jest.fn();
  const spyOnBlurHandler = jest.fn();
  const value = 'text value';
  const disabled = false;
  const required = true;
  let error = false;
  let errorHelperText = '';

  let input: RenderResult;

  beforeEach(() => {
    input = render(
      <InputField
        type={type}
        label={label}
        name={name}
        onChange={spyOnChangeHandler}
        onBlur={spyOnBlurHandler}
        value={value}
        disabled={disabled}
        required={required}
        error={error}
        errorHelperText={errorHelperText}
      />
    );
  });

  it('tests text type input field renders correctly', () => {
    const { getByTestId, getByText } = input;
    const fieldLabel = getByText(label);
    const field = getByTestId(inputFieldTestID)
      .querySelector('input[type="text"]') as HTMLInputElement;

    expect(fieldLabel).toBeInTheDocument();
    expect(field).toHaveProperty('type', type);
    expect(field).toHaveProperty('name', name);
    expect(field).toHaveProperty('value', value);
  });

  it('tests password input type rendered', () => {
    type = 'password';
    const { rerender, getByTestId } = input;
    rerender(<InputField
      type={type}
      label={label}
      name={name}
      onChange={spyOnChangeHandler}
      onBlur={spyOnBlurHandler}
      value={value}
      disabled={disabled}
      required={required}
      error={error}
      errorHelperText={errorHelperText}
    />);
    const field = getByTestId(inputFieldTestID)
      .querySelector('input[type="password"]') as HTMLInputElement;
    expect(field).toHaveProperty('type', type);
  });

  it('tests number input type rendered', () => {
    type = 'number';
    const { rerender, getByTestId } = input;
    rerender(<InputField
      type={type}
      label={label}
      name={name}
      onChange={spyOnChangeHandler}
      onBlur={spyOnBlurHandler}
      value={value}
      disabled={disabled}
      required={required}
      error={error}
      errorHelperText={errorHelperText}
    />);
    const field = getByTestId(inputFieldTestID)
      .querySelector('input[type="number"]') as HTMLInputElement;
    expect(field).toHaveProperty('type', type);
  });

  it('tests type input field error state', () => {
    error = true;
    errorHelperText = 'error helper text';
    const { rerender, getByText } = input;
    rerender(<InputField
      type={type}
      label={label}
      name={name}
      onChange={spyOnChangeHandler}
      onBlur={spyOnBlurHandler}
      value={value}
      disabled={disabled}
      required={required}
      error={error}
      errorHelperText={errorHelperText}
    />);
    const errorText = getByText(errorHelperText);
    expect(errorText).toBeInTheDocument();
  });

  it('tests callbacks', () => {
    const { getByTestId } = input;
    const inputField = getByTestId(inputFieldTestID)
      .querySelector('input[type="number"]') as HTMLInputElement;

    fireEvent.change(inputField, { target: { value: 'test value' } });
    expect(spyOnChangeHandler).toHaveBeenCalledTimes(1);

    fireEvent.blur(inputField);
    expect(spyOnBlurHandler).toHaveBeenCalledTimes(1);
  });

  it('tests if matches with snapshot', () => {
    const { asFragment } = input;
    expect(asFragment()).toMatchSnapshot();
  });
});