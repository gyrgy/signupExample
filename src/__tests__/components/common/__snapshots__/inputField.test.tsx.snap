// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test input field component tests if matches with snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    data-testid="components/constants/form"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-error Mui-error Mui-required Mui-required"
      data-shrink="false"
    >
      label
      <span
        class="MuiFormLabel-asterisk MuiInputLabel-asterisk Mui-error Mui-error"
      >
         *
      </span>
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error MuiInputBase-fullWidth MuiInputBase-formControl"
    >
      <input
        aria-invalid="true"
        class="MuiInputBase-input MuiOutlinedInput-input"
        name="name"
        required=""
        type="number"
        value="text value"
      />
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-385 MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="PrivateNotchedOutline-legendLabelled-387"
        >
          <span>
            label *
          </span>
        </legend>
      </fieldset>
    </div>
    <p
      class="MuiFormHelperText-root MuiFormHelperText-contained Mui-error Mui-required"
    >
      error helper text
    </p>
  </div>
</DocumentFragment>
`;
